File: Queue/Implementation in C/Queue.c

#include <stdio.h>
#include <stdbool.h>
#define capacity 10

typedef struct{
    int front;
    int rear;
    int capacity;
    int array[capacity];
}Queue;

// Queue Basic Functions
// Queue createQueue function to creat the Queue
void createQueue(&queue){
    queue->front = 0;
    queue->rear = -1;
    queue->size = 0;
}

// Queue isFull function to check if queue is full
bool isFull(&queue){
    return queue->size = capacity;
}

// Queue isEmpty function to check if queue is empty
bool isEmpty(&queue){
    return size == 0;
}

// Queue enqueue function that add element in the last index
void enqueu(int data, &queue){
    if (isFull()) {
		printf("Queue overflow.\n");
		return;
	}
    
    queue->rear++;
    queue->array[rear] = data;
    queue->size++;
}

// Queue dequeue function that returns and remove the first element
void dequeue(&queue){
    if (isEmpty()) {
		printf("Queue underflow.\n");
		return;
	}
	
	int data = array[front];
	queue->front++;
	queue->size--;
	
	return data;
}

// Queue clear function to clear the Queue
void clear(&queue){
    queue->front = queue->size = 0;
    queue->rear = -1;
}

// Queue peek function that returns the first(front) element
void peek(&queue){
    if (isEmpty())
		return;
	
    return queue->array[front];
}



// Queue printQueue function that prints all Queue elements
void printQueue(&queue){
    if (isEmpty())
		return;
		
	for (int i = queue->front; i <= queue->rear; i++)
		printf("[%d] -> %d", i, queue->array[i]);
}


int main()
{
    printf("Hello World");

    return 0;
}

File: Queue/Implementation in Java/Queue.java
package mypackage;


// Implementation of the Queue
public class Queue {
	protected int[] arr;
	protected int capacity;
	protected int front;
	protected int rear;
	protected int size;

	// Constructor that defines the Queue
	public Queue(int capacity) {
		this.capacity = capacity;
		arr = new int[capacity];
		front = 0;
		rear = -1;
		size = 0;
	}


	// Queue Basic Functions
	// Queue enqueue function that add element in the last index
	public void enqueue(int data) {
		if (isFull()) {
			System.out.println("Queue overflow.");
			return;
		}

		rear++;
		arr[rear] = data;
		size++;
	}

	// Queue dequeue function that returns and remove the first element
	public int dequeue() {
		if (isEmpty()) {
			System.out.println("Queue underflow.");
			return -1;
		}

		int data = arr[front];
		front++;
		size--;
		return data;
	}

	// Queue clear function to clear the Queue
	public void clear() {
		front = size = 0;
		rear = -1;
	}

	// Queue peek function that returns the first(front) element
	public int peek() {
		if (isEmpty())
			return -1;

		return arr[front];
	}

	// Check if queue is full
	public boolean isFull() {
		return size == capacity;
	}

	// Check if queue is empty
	public boolean isEmpty() {
		return size == 0;
	}


	// Queue printQueue function that prints all Queue elements
	public void printQueue() {
		if (isEmpty())
			return;

		System.out.print("Queue elements: ");
		for (int i = front; i <= rear; i++) {
			System.out.print(arr[i] + " ");
		}
		System.out.println();
	}
}
